Mono.Nat/AsyncExtensions.cs

```diff
@@ -57,12 +57,31 @@ public static async void FireAndForget (this Task task)
         public static void WaitAndForget (this Task task)
         {
             try {
-                task.GetAwaiter ().GetResult ();
+                task.ConfigureAwait (false).GetAwaiter().GetResult();
```

Comment:
I'm pretty certain this is a no-op as we are calling GetResult synchronously.

Either way, this method is being removed so it won't be a problem in the end.

Reply:
I do believe this was the only line the debugger showed me it was stopping on, and I was just trying to make sure it wasn't stopped because of a deadlock.  I'm at beginner level with async / await in C#, and have a poor understanding of how it works internally.  I still find it baffling that Microsoft would build this asynchronous programming system where deadlocks are so extremely likely if you don't call `ConfigureAwait (false)` all over the place.

Reply:
I agree. What I would add is that `task.GetAwaiter ().GetResult ();` is a *very* poor programming practice and it was a bit lazy of me to do this to begin with. You can make it highly unlikely for this pattern to deadlock if you write a custom task scheduler/sync context object, but it's nearly never worth the effort unless you have a large synchronous codebase which you can't modernise.

MS have https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.joinabletaskfactory?view=visualstudiosdk-2019 for this purpose, though if you have a practical use for this then you already have an unsolvable problem in your applications architecture :)